langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockAnthropicMessageChatModel.java

```diff
@@ -53,47 +69,259 @@ protected Map<String, Object> getRequestParameters(String prompt) {
+            propertyNode.set("enum", enumValues);
+        }
+
+        return propertyNode;
+    }
+
     private List<BedrockAnthropicMessage> getAnthropicMessages(List<ChatMessage> messages) {
-        return messages.stream()
-            .filter(message -> message.type() != ChatMessageType.SYSTEM)
-            .map(message -> new BedrockAnthropicMessage(getAnthropicRole(message), getAnthropicContent(message)))
-            .collect(Collectors.toList());
+        List<ChatMessage> noSystemMessages = messages.stream()
+                .filter(message -> message.type() != ChatMessageType.SYSTEM)
+                .collect(Collectors.toList());
+
+        List<BedrockAnthropicMessage> anthropicMessages = new ArrayList<>();
+        List<BedrockAnthropicContent> toolContents = new ArrayList<>();
+
+        for (ChatMessage message : noSystemMessages) {
+            List<BedrockAnthropicContent> contents = getAnthropicContent(message);
+
+            if (message instanceof ToolExecutionResultMessage) {
+                toolContents.addAll(getAnthropicContent(message));
+                continue;
+            } else {
+                if (!toolContents.isEmpty()) {
+                    anthropicMessages.add(new BedrockAnthropicMessage("user", toolContents));
+                    toolContents = new ArrayList<>();
+                }
+
+                if (message instanceof UserMessage) {
+                    anthropicMessages.add(new BedrockAnthropicMessage("user", contents));
+                }
+
+                if (message instanceof AiMessage) {
+                    anthropicMessages.add(new BedrockAnthropicMessage("assistant", contents));
+                }
+            }
+        }
+
+        if (!toolContents.isEmpty()) {
+            anthropicMessages.add(new BedrockAnthropicMessage("user", toolContents));
+        }
+
+        return anthropicMessages;
     }
-    
+
     private List<BedrockAnthropicContent> getAnthropicContent(ChatMessage message) {
         if (message instanceof AiMessage) {
-            return Collections.singletonList(new BedrockAnthropicContent("text", ((AiMessage) message).text()));
+            AiMessage aiMessage = (AiMessage) message;
+            List<BedrockAnthropicContent> contents = new ArrayList<>();
+
+            if (isNotNullOrBlank(aiMessage.text())) {
+                contents.add(new BedrockAnthropicContent("text", aiMessage.text()));
+            }
+
+            if (aiMessage.hasToolExecutionRequests()) {
+                List<BedrockAnthropicContent> toolUseRequests = aiMessage.toolExecutionRequests().stream()
+                        .map(toolExecutionRequest -> BedrockAnthropicContent.builder()
+                                .id(toolExecutionRequest.id())
+                                .type("tool_use")
+                                .name(toolExecutionRequest.name())
+                                .input(Json.fromJson(toolExecutionRequest.arguments(), Map.class))
```

Comment:
Please do not use `Json.fromJson()`, use `ObjectMapper`

Reply:
Done.